### ISSUE TITLE
Backup stalled on "finishing" on mdadm RAID5

### ISSUE THREAD

**Issue by adocampo** [grke/burp#858]
I'm having issues on a brand new installation of burp server 2.3.32 (tried with both Debian 10 and Ubuntu 18.04).

I can do the very first backup without problems, but the second one stalls on "finishing" after the successful copy of files. It cannot then start a third backup as it says there is another backup in progress, which is false. I can manually delete the finishing link and point current link to the latest backup, and then the following backup would start but stuck as well on finishing.

I have a RAID5 array mounted the /var/spool/burp, but when I unmount the RAID and I use my rootfs single SSD, it works as expected.

On that RAID5, I have XFS with default options.

I'm really out of ideas, I always used burp with glusterfs over zfs with no problems at all, or inside a VM with a single xfs formatted qcow2 disk (qcow disk over zfs+glusterfs), so that's really the very first time I tried xfs over RAID5 and this issue is driving me crazy.

Any tip?

**Comment by grke** [comment]
Hello,

When a backup is in 'finishing' state, you cannot just manually delete the 'finishing' link and repoint 'current' without risking breaking everything, because the process of converting the 'finishing' to 'current' changes the previous 'current'. Once it is 'finishing', you should only go forwards.

I recommend starting from scratch, and if you get stuck in 'finishing' again, check the server to see if it really is doing nothing. Have a look in the server side client-specific logs in the backup itself, to see if it is logging anything useful.

**Comment by adocampo** [comment]
I finally destroyed that RAID5 and created a ZFS raidz1 in its place. Backup went way faster (we're speaking about 13 GB of the local burp-server itself where on RAID5 took almost 10 minutes in phase 1 and 2, and just 1 minute 10'' with ZFS) and it finally managed to finishing all the phases after three minutes. So it's definitively happening something with XFS+mdadm RAID5, I guess it finally would finish all the phase but at a damn slow speed (first time I tried with Ubuntu, I let it a whole weekend and it didn't finish).

As this "bare metal" server is hosted on OVH, I sincerely doubt about the quality of the disks or if it is really a physical and not some kind of semi-virtualized host with iscsi disks (as the baremetal server is deployed and ready to use in 120s since the purchase time). So I'm going to use ZFS as at least it's response time is acceptable. We can close this as I don't really think this can be a bug, but just a combination of situations and too many unknowns regarding the hardware.

Thank you @grke!
